apps/trading/e2e/actions/utils.py

```diff
@@ -35,4 +35,29 @@ def next_epoch(vega: VegaServiceNull):
                 "Epoch not started after forwarding the duration of two epochs."
             )
     vega.wait_fn(1)
-    vega.wait_for_total_catchup()
\ No newline at end of file
+    vega.wait_for_total_catchup()
```

Comment:
not addressable but I would love to see us put a more sure-fire wait mechanism in here. From what you have said yourself the wait_fn seems a little hopeful in places. Would be good to take a look at some point soon